src/controller/python/chip/icd/PyChipCheckInDelegate.h

```diff
@@ -0,0 +1,40 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/icd/client/DefaultCheckInDelegate.h>
+
+class PyChipCheckInDelegate : public chip::app::DefaultCheckInDelegate
+{
+public:
+    using OnCheckInCompleteCallback = void(uint8_t fabricIndex, uint64_t nodeId);
+
+    virtual ~PyChipCheckInDelegate() = default;
+
+    void OnCheckInComplete(const chip::app::ICDClientInfo & clientInfo) override;
+
+    void SetOnCheckInCompleteCallback(OnCheckInCompleteCallback * callback) { mCallback = callback; }
+
+    static PyChipCheckInDelegate & GetInstance() { return sInstance; }
```

Comment:
Is there a reason this must be a singleton? Is there no clean way to inject a single instance without needing to access globally?

Reply:
yes, i think we can initialize this inside ChipDeviceController-ScriptDevicePairingDelegate.cpp

Reply:
Done by moving to ChipDeviceController-ScriptBinding.cpp

I think we currently don't have a global objects holder for python, but anyway the global object are at a single place for now.